src/grpc_requests/aio.py

```diff
@@ -477,8 +495,20 @@ async def _register_file_descriptor(self, file_descriptor):
             )
             for dep_file_name in dependencies:
                 if not self._is_descriptor_registered(dep_file_name):
-                    dep_desc = await self.get_file_descriptor_by_name(dep_file_name)
-                    await self._register_file_descriptor(dep_desc)
+                    # First look for dependency in the passed in descriptors
+                    dep_desc = next(
+                        (x for x in file_descriptors if x.name == dep_file_name), None
+                    )
```

Comment:
If the goal here is simply to determine whether or not the dependency description is available in the list of existing file descriptors - do you strictly need to build the iterator, or can you simply use the set comprehension and call your follow up logic if the result is the empty set?

Reply:
I'll give that a shot

Reply:
Good thought - hadn't considered the short-circuit working out that way. Sounds like the way to go then!